packages/pwa/app/hooks/use-site.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {pathToUrl} from '../utils/url'
+import {useLocation} from 'react-router-dom'
+import {resolveSiteFromUrl} from '../utils/site-utils'
+import {useMemo} from 'react'
+
+/**
+ * This hook returns the current site based on current location
+ *
+ * @returns {Object} - current site
+ */
+const useSite = () => {
```

Comment:
_Pasted from slack_

[@Alex Vuong](https://mobify.slack.com/team/UKVSJEY0M)
 with respect to the comment about naming the hook useSiteConfig  vs useSite this one is a little tricky. We really have to ask ourselves what a site is, 
[@Vincent Marta](https://mobify.slack.com/team/U0DKKM9JN)
 is suggesting that the object, because it’s in the config file, is actually a site config type object. But we know that our intention is to load that data via an API eventually and I believe the types of the object returned are of site type. So if’s thats the case, useSite is probably correct. The tricky part is that we are then left trying to combine site id and site alias into one variable name. We can do that but we have to some up with a nice name. Maybe we have siteId, siteAlias and siteRef where it’s value is the alias or id if none is defined.